packages/compiler-sfc/src/compileScript.ts

```diff
@@ -1494,6 +1557,17 @@ export function compileScript(
         emitsTypeDecl.end!
       )}), expose: any, slots: any, attrs: any }`
     }
+
+    // TODO review this part
+    if (slotsTypeDecl) {
+      const content = slotsTypeDecl.__fromNormalScript
+        ? script!.content
+        : scriptSetup.content
+      args += `: { slots: (${content.slice(
+        slotsTypeDecl.start!,
+        slotsTypeDecl.end!
+      )}), expose: any, emit: any, attrs: any }`
+    }
```

Comment:
Not sure about this, this seems that might cause an issue when we have both slot and emit